allauth/account/forms.py

```diff
@@ -183,6 +183,9 @@ def clean(self):
                 login = self.cleaned_data["login"]
                 if self._is_login_email(login):
                     auth_method = app_settings.AuthenticationMethod.EMAIL
+                    email = user.email
+                    if EmailAddress.objects.filter(email=email, verified=False).exists() is True and app_settings.LOGIN_VERIFIED_EMAIL_ONLY is True:
```

Comment:
@9mido wouldn't it be more efficient to put the settings check first? Database-access-wise I mean

Reply:
You mean change the order of the if statement to be:

`if app_settings.LOGIN_VERIFIED_EMAIL_ONLY is True and EmailAddress.objects.filter(email=email, verified=False).exists() is True:` ? I don't think it really matters much.

Reply:
Access to settings does not require a db load so first checking that would reduce dB actions. But it is marginal

Reply:
Right. Can you show me the exact code correction you would like to see and how I would change it?

Reply:
if app_settings.LOGIN_VERIFIED_EMAIL_ONLY is True and EmailAddress.objects.filter(email=email, verified=False).exists() is True: